### PR TITLE
rand score instead of accuracy

### PR DESCRIPTION
**Pull request by Farnazmdi**
`balanced_accuracy_score` is now replaced with `rand_score`. 
The function `permute_states()` does the state switching and is called inside the `Results()`.
I included some of JC's edits from the other PR into this to avoid conflicts as we were working on the same files.
I have not written a test function yet, but I am sure that the functions are working fine. I created some of the figures myself and saw the performance.
I wanted to ask JC to recreate all of the figures after we merge this. 
Please let me know if it is alright to do that or you want figures recreated in this PR.

Thank you.

### CODE REVIEW

**Review by aarmey — CHANGES_REQUESTED** [review]
These change some of the figure files. Should the figure svgs be removed so they build?

**Comment by Farnazmdi** [comment]
Oh.. you are right.. I was going to ask JC to rebuild the figures, but I can just remove them and they will be built!!
Thanks, will do right away.

File: lineage/Analyze.py [hunk 1]
```diff
@@ -173,3 +137,45 @@ def run_Results_over(output: list, parallel=True) -> list:
 
     prom_holder = [exe.submit(Results, *x) for x in output]
     return [prom.result() for prom in prom_holder]
+
+def permute_states(tHMMobj: Any, pred_states: list) -> Tuple[Any, list]:
```
Comment (aarmey) [inline lineage/Analyze.py:142]: So if I'm following, the true states are stored in `tHMMobj.X.output_lineage`. If the tHMM object is fit, and has the true values stored in it, why do we need `pred_states`? Can't the predicted states be generated by calling Viterbi using the data (which is stored inside the tHMM object)? It'd be nice to get rid of a redundant argument if possible.

File: lineage/Analyze.py [hunk 2]
```diff
@@ -173,3 +137,45 @@ def run_Results_over(output: list, parallel=True) -> list:
...
+    """
+    This function takes the tHMMobj and the predicted states,
+    and finds out whether we need to switch the state identities or not based on the likelihood.
+    """
+    true_states = [[cell.state for cell in lineage.output_lineage] for lineage in tHMMobj.X]
```
Comment (aarmey) [inline lineage/Analyze.py]: `true_states` is never used in here.

File: lineage/Analyze.py [hunk 3]
```diff
@@ -91,68 +91,32 @@ def Results(tHMMobj, pred_states_by_lineage: list, LL: float) -> dict[str, Any]:
     # Instantiating a dictionary to hold the various metrics of accuracy and scoring for the results of our method
     results_dict: dict[str, Any]
     results_dict = {}
+    tHMMobj, pred_states = permute_states(tHMMobj, pred_states_by_lineage)
```
Comment (aarmey) [inline lineage/Analyze.py:94]: Here again, does Results() need `pred_states_by_lineage`, or can it just retrieve this my `tHMM.predict()`?

File: lineage/figures/figure4.py [hunk 4]
```diff
@@ -114,16 +114,16 @@ def figureMaker3(ax, x_Sim, output_Sim, x_Cen, output_Cen, xlabel="Number of Cel
 
     i += 1
     ax[i].axhline(y=100, ls='--', c='k', alpha=0.5)
-    sns.regplot(x="Cell number", y="State Assignment Accuracy", data=accuracy_sim_df, ax=ax[i], lowess=True, marker='+', scatter_kws=scatter_state_1_kws)
+    sns.regplot(x="Cell number", y="State Assignment Performance", data=accuracy_sim_df, ax=ax[i], lowess=True, marker='+', scatter_kws=scatter_state_1_kws)
     ax[i].set_xlabel(xlabel)
-    ax[i].set_ylim(bottom=0, top=101)
+    ax[i].set_ylim(bottom=50, top=101)
```
Comment (aarmey) [inline lineage/figures/figure4.py:119]: Did the reviewers ask for these axis changes? I'm not generally a fan of y-axis that don't extend to the natural limit (in this case 0).

**Review by aarmey — CHANGES_REQUESTED** [review]

File: lineage/Analyze.py (hunk 3 above)
Comment (Farnazmdi) [reply lineage/Analyze.py:94]: This needs editing a couple of things, specifically, the output of `Analyze()` is linked to this, is it alright if we leave it this way? 
I changed the `permute_states()` and it only takes in the `tHMMobj`.

File: lineage/figures/figure4.py (hunk 4 above)
Comment (Farnazmdi) [reply lineage/figures/figure4.py:119]: Only for Bernoulli parameter estimation. Although the reviewer did ask for changing the yaxis if figure 4 (censored vs uncensored) but I will reverse that.

**Comment by Farnazmdi** [comment]
I also updated the figures.

**Comment by aarmey** [comment]
Remind me, what is the benefit of committing all the svg files?

File: lineage/figures/figure6.py [hunk 5]
```diff
@@ -88,13 +87,13 @@ def figureMaker5(ax, distribution_df, wasser_df):
...
 
-    sns.violinplot(x="G1 lifetime", y="Distribution similarity", hue="State", palette={"State 1": "b", "State 2": "g"}, split=True, data=distribution_df, ax=ax[i])
+    sns.violinplot(x="G1 lifetime", y="Distribution Similarity", hue="State", palette={"State 1": "b", "State 2": "g"}, split=True, data=distribution_df, ax=ax[i])
 
     i += 1
     # state accuracy
-    sns.regplot(x="Wasserstein distance", y="State Assignment Accuracy", data=wasser_df, label="tHMM", ax=ax[i], lowess=True, marker='+')
-    sns.regplot(x="Wasserstein distance", y="KMeans accuracy", data=wasser_df, ax=ax[i], label="K-means", lowess=True, marker='+')
-    ax[i].set_title("State Assignment Accuracy")
+    sns.regplot(x="Wasserstein distance", y="State Assignment Performance", data=wasser_df, label="tHMM", ax=ax[i], lowess=True, marker='+')
```
Comment (aarmey) [inline lineage/figures/figure6.py:94]: Accuracy seems like a better term than Performance, which is too vague. To be more specific, this could be "Random Index Accuracy" or some such.

File: lineage/tests/test_Viterbi.py [hunk 6]
```diff
@@ -1,6 +1,7 @@
 """ Unit test file for Viterbi. """
 import unittest
-from numpy.random import randint
+import numpy as np
+from sklearn.metrics import rand_score
```
Comment (aarmey) [inline lineage/tests/test_Viterbi.py:4]: Doesn't look like `rand_score` is used here.

**Review by aarmey — CHANGES_REQUESTED** [review]

**Comment by Farnazmdi** [comment]
> Remind me, what is the benefit of committing all the svg files?

Just that the manuscript is updated. I can download them separately and put them in a file for resubmission.

File: lineage/figures/figure6.py (hunk 5 above)
Comment (Farnazmdi) [reply lineage/figures/figure6.py:94]: Will replace them with "Random Index Accuracy". Thanks.

**Comment by Farnazmdi** [comment]
@aarmey do you mind taking a look at this error in the tests? I thought it would create the cache folder each time..
image

**Comment by aarmey** [comment]
Don't know why it's failing. I just removed the mypy check for now.

File: lineage/figures/figure11.py [hunk 7]
```diff
@@ -30,3 +30,5 @@ def makeFigure():
     return f
 
 # plot_networkx(T_lap.shape[0], T_lap, 'lapatinib')
+
+def
```
Comment (aarmey) [inline lineage/figures/figure11.py:34]: What is this?

File: lineage/figures/figure11.py [hunk 8]
```diff
@@ -30,3 +30,5 @@ def makeFigure():
     return f
 
 # plot_networkx(T_lap.shape[0], T_lap, 'lapatinib')
```
Comment (aarmey) [inline lineage/figures/figure11.py:32]: Remove commented out line.

File: lineage/figures/figure8.py [hunk 9]
```diff
@@ -104,11 +103,11 @@ def figure_maker(ax, BIC_Holder, true_state_no, upper_ylim, censored=False):
     BIC_holder = list(map(list, zip(*BIC_Holder)))
 
     # Creating Histogram and setting ylim
-    ax2 = ax.twinx()
-    ax2.hist(np.argmin(BIC_holder, axis=0) + 1, rwidth=1,
-             alpha=.2, bins=desired_num_states, align='left')
-    ax2.set_yticklabels([])
-    ax2.set_yticks([])
+    # ax2 = ax.twinx()
+    # ax2.hist(np.argmin(BIC_holder, axis=0) + 1, rwidth=1,
```
Comment (aarmey) [inline lineage/figures/figure8.py:107]: Remove, not comment.

File: lineage/figures/figureCommon.py [hunk 10]
```diff
@@ -17,7 +17,7 @@
 from ..states.StateDistributionGaPhs import StateDistribution as phaseStateDist
 
 # T: transition probability matrix
-T = np.array([[0.9, 0.1], [0.05, 0.95]], dtype="float")
+T = np.array([[0.9, 0.1], [0.1, 0.9]], dtype="float")
```
Comment (aarmey) [inline lineage/figures/figureCommon.py:20]: Confirm that this is updated in the text.

**Review by aarmey — CHANGES_REQUESTED** [review]
Some changes.

In Fig. S1, why are k, theta, and p in a different font from the rest of the text? Also, it seems strange to have "Random Index Accuracy" as the title when it is the y-axis. The title can be "State Assignment Accuracy" and the y-axis "Random Index Accuracy".

File: lineage/figures/figureCommon.py (hunk 10 above)
Comment (Farnazmdi) [reply lineage/figures/figureCommon.py:20]: It is updated.

**Comment by Farnazmdi** [comment]
> Some changes.
> 
> In Fig. S1, why are k, theta, and p in a different font from the rest of the text? Also, it seems strange to have "Random Index Accuracy" as the title when it is the y-axis. The title can be "State Assignment Accuracy" and the y-axis "Random Index Accuracy".

They are addressed now. For the fonts, I made `Arial` the default font, so I guess it should be fixed.
Thank you.

**Comment by aarmey** [comment]
Merging!
